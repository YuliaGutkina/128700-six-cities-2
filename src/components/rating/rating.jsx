import React from 'react';
import PropTypes from "prop-types";
import classNames from "classnames";


export const Rating = (props) => {
  const {value, className, starsClassName, valueClassName, isValueVisible} = props;

  return <div className={classNames(`rating`, className)}>
    <div className={classNames(`rating__stars`, starsClassName)}>
      <span style={{width: `${Math.round(value) * 20}%`}} />
      <span className="visually-hidden">Rating</span>
    </div>
    {isValueVisible &&
      <span className={classNames(`rating__value`, valueClassName)}>{value}</span>
    }
  </div>;
};

Rating.propTypes = {
  value: PropTypes.number.isRequired,
  className: PropTypes.string,
  valueClassName: PropTypes.string,
  starsClassName: PropTypes.string,
  isValueVisible: PropTypes.bool,
};
