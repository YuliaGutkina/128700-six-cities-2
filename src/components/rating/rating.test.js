import React from "react";
import renderer from "react-test-renderer";
import {Rating} from "./rating";


it(`Rating correctly renders after relaunch`, () => {
  const tree = renderer
    .create(
        <Rating value={3.2}/>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
