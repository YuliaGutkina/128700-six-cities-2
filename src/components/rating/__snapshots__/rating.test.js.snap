// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating correctly renders after relaunch 1`] = `
<div
  className="rating"
>
  <div
    className="rating__stars"
  >
    <span
      style={
        Object {
          "width": "60%",
        }
      }
    />
    <span
      className="visually-hidden"
    >
      Rating
    </span>
  </div>
</div>
`;
